File: python/Figuras/figura.py
"""
Clase base abstracta Figura
"""
# pylint: disable=invalid-name
from abc import ABC

class Figura(ABC):
    """Clase abstracta base para todas las figuras geométricas"""

    def __init__(self, nombre: str, tipo: str, id_figura: int):
        """
        Constructor de la clase Figura

        Args:
            nombre (str): Nombre de la figura
            tipo (str): Tipo de figura
            id_figura (int): Identificador único de la figura
        """
        self._nombre = nombre
        self._tipo = tipo
        self._id_figura = id_figura

    def get_nombre(self) -> str:
        """Obtiene el nombre de la figura"""
        return self._nombre

    def get_tipo(self) -> str:
        """Obtiene el tipo de la figura"""
        return self._tipo

    def get_id(self) -> int:
        """Obtiene el ID de la figura"""
        return self._id_figura

    def set_nombre(self, nombre: str) -> None:
        """Establece el nombre de la figura"""
        self._nombre = nombre

    def set_tipo(self, tipo: str) -> None:
        """Establece el tipo de la figura"""
        self._tipo = tipo

File: java/Figuras/src/main/java/org/example/Figura.java
package org.example;

/**
 * Clase base abstracta Figura
 */
public abstract class Figura {
    
    private String nombre;
    private String tipo;
    private int idFigura;
    
    /**
     * Constructor de la clase Figura
     * @param nombre Nombre de la figura
     * @param tipo Tipo de figura
     * @param idFigura Identificador único de la figura
     */
    public Figura(String nombre, String tipo, int idFigura) {
        this.nombre = nombre;
        this.tipo = tipo;
        this.idFigura = idFigura;
    }
    
    /**
     * Obtiene el nombre de la figura
     * @return El nombre de la figura
     */
    public String getNombre() {
        return this.nombre;
    }
    
    /**
     * Obtiene el tipo de la figura
     * @return El tipo de la figura
     */
    public String getTipo() {
        return this.tipo;
    }
    
    /**
     * Obtiene el ID de la figura
     * @return El ID de la figura
     */
    public int getId() {
        return this.idFigura;
    }
    
    /**
     * Establece el nombre de la figura
     * @param nombre El nuevo nombre
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    /**
     * Establece el tipo de la figura
     * @param tipo El nuevo tipo
     */
    public void setTipo(String tipo) {
        this.tipo = tipo;
    }
}